internal/controllers/machine/drain/filters.go

```diff
@@ -275,6 +286,11 @@ func (d *Helper) drainLabelFilter(ctx context.Context, pod *corev1.Pod) PodDelet
 		log.V(4).Info(fmt.Sprintf("Skip evicting Pod, because Pod has %s label", clusterv1.PodDrainLabel))
 		return MakePodDeleteStatusSkip()
 	}
+	if labelValue, found := pod.ObjectMeta.Labels[clusterv1.PodDrainLabel]; found && strings.EqualFold(labelValue, string(clusterv1.MachineDrainRuleDrainBehaviorWaitCompleted)) {
```

Comment:
I think we also have to update drain.go l.228 to break for MachineDrainRuleDrainBehaviorWaitCompleted. Otherwise the filter result will be overwritten by subsequent filters (in this case the machineDrainRulesFilter)